File: SDKs/amazonaws.com/cur/2017-01-06/python/src/sdk/models/shared/putreportdefinitionrequest.py
from dataclasses import dataclass, field
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class PutReportDefinitionRequest:
    r"""PutReportDefinitionRequest
    Creates a Cost and Usage Report.
    """
    
    report_definition: ReportDefinition = field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('ReportDefinition') }})
    

File: SDKs/amazonaws.com/cur/2017-01-06/typescript/src/sdk/models/shared/putreportdefinitionrequest.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { ReportDefinition } from "./reportdefinition";



// PutReportDefinitionRequest
/** 
 * Creates a Cost and Usage Report.
**/
export class PutReportDefinitionRequest extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=ReportDefinition" })
  reportDefinition: ReportDefinition;
}
